### PR TITLE
OEL-740: Banner paragraph kernel testing.

### PR DESCRIPTION
**Pull request by escuriola**
Jira issue(s):
- https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-740

### CODE REVIEW

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_bootstrap_theme\Kernel\Paragraphs;
+
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Tests the rendering of paragraph types with media fields.
+ *
+ * @group batch1
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: remove this, we dont have batches set up

File: tests/src/Kernel/Paragraphs/BasePatternAssert.php [hunk 2]
```diff
@@ -0,0 +1,201 @@
+<?php
+
+declare(strict_types = 1);
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/BasePatternAssert.php]: this is a bit confusing for me, can we move these in their own folder outside the paragraphs

File: tests/src/Kernel/Paragraphs/PatternAssertInterface.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?php
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/PatternAssertInterface.php]: move in its own folder

File: tests/src/Kernel/Paragraphs/TextFeaturedMediaAssert.php [hunk 4]
```diff
@@ -0,0 +1,139 @@
+<?php
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/TextFeaturedMediaAssert.php]: move to own folder

File: tests/src/Kernel/AbstractKernelTestBase.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    'ui_patterns_settings',
+    'ui_patterns_library',
+    'user',
+    'node',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
```
Comment (drishu) [inline tests/src/Kernel/AbstractKernelTestBase.php]:
```suggestion
  protected function setUp(): void {
```

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+    'oe_media_avportal',
+    'options',
+    'oe_media_iframe',
+    'file_link',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]:
```suggestion
  protected function setUp(): void {
```

File: tests/src/Kernel/Paragraphs/ParagraphsTestBase.php [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    'oe_paragraphs',
+    'allowed_formats',
+    'locale',
+    'node',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/ParagraphsTestBase.php]:
```suggestion
  protected function setUp(): void {
```

File: tests/src/Kernel/Paragraphs/ParagraphsTestBase.php [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+/**
+ * Base class for paragraphs tests.
+ */
+abstract class ParagraphsTestBase extends AbstractKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/ParagraphsTestBase.php]:
```suggestion
  protected static $modules = [
```

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 9]
```diff
@@ -0,0 +1,482 @@
...
+ * Tests the rendering of paragraph types with media fields.
+ *
+ * @group batch1
+ */
+class MediaParagraphsTest extends ParagraphsTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: there is no point to declare a $modules in ParagraphsTestBase and overwrite the complete list here, remove this and add in ParagraphsTestBase::modules the complete list of what you need (and only the ones you need)

File: tests/src/Kernel/Paragraphs/MediaParagraphsTest.php [hunk 10]
```diff
@@ -0,0 +1,482 @@
...
+    $paragraph->get('field_oe_banner_type')->setValue('hero_left');
+    $paragraph->save();
+
+    // Unpublish the media and assert it is not rendered anymore.
+    $media->set('status', 0);
+    $media->save();
+
+    // Since static cache is not cleared due to lack of requests in the test we
+    // need to reset manually.
+    $this->container->get('entity_type.manager')->getAccessControlHandler('media')->resetCache();
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/MediaParagraphsTest.php]: no point in flushing before, we have the one after

File: tests/src/Kernel/Paragraphs/TextFeaturedMediaAssert.php [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+
+use Symfony\Component\DomCrawler\Crawler;
+use PHPUnit\Framework\Exception;
+
+/**
+ * Assertions for text_featured_media pattern.
+ *
+ *  @see ./templates/patterns/text_featured_media/text_featured_media.ui_patterns.yml
+ */
+class TextFeaturedMediaAssert extends BasePatternAssert {
```
Comment (drishu) [inline tests/src/Kernel/Paragraphs/TextFeaturedMediaAssert.php]: where is this class used ?

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — APPROVED** [review]

**Review by GilNovacomm — APPROVED** [review]

**Pull request merged**